lib/src/fnc/script/fetch/classes/request.rs

```diff
@@ -395,6 +402,16 @@ impl<'js> Request<'js> {
 			let url_str = url.to_string()?;
 			let url = Url::parse(&url_str)
 				.map_err(|e| Exception::throw_type(&ctx, &format!("failed to parse url: {e}")))?;
+
+			// Check if the url is allowed
+			let query_ctx = ctx
+				.globals()
+				.get::<_, OwnedBorrow<'js, QueryContext<'js>>>(QUERY_DATA_PROP_NAME)?;
+			query_ctx
+				.context
+				.check_allowed_net(&url)
+				.map_err(|e| Exception::throw_message(&ctx, &e.to_string()))?;
```

Comment:
Do we want to limit creating request objects with disallowed urls? 

Personally I would expect an error to be throw when I try to actually send the request by calling fetch on the object.

Reply:
Agreed. I was testing my implementation based on this code:
```
RETURN function() {
  return fetch("https://www.surrealdb.com");
}
```

Could you provide an example of creating the Request object without using it so I can test it and change the implementation accordingly? Where is the `fetch` actually performed?

Reply:
The actual request is made here https://github.com/surrealdb/surrealdb/blob/76445cc4dd5d5a35168bdb645f44c673090a7424/lib/src/fnc/script/fetch/func.rs#L94 

The url for the request can be found here: https://github.com/surrealdb/surrealdb/blob/76445cc4dd5d5a35168bdb645f44c673090a7424/lib/src/fnc/script/fetch/func.rs#L28

and a basic example of using a request object without sending it:
```js
RETURN function(){
    let _req = new Request("https://www.suurealdb.com");
    // You would use it as following:
    // await fetch(_req);
}
```

Reply:
thanks @DelSkayn 🙏